### PR TITLE
Support php versions from 5.7 to 7.4

### PR DESCRIPTION
**Pull request by everzet**
Primarily a rebase of #26 and removes the need in #27

### CODE REVIEW

**Comment by snapshotpl** [comment]
@everzet looks good. 🛳

**Comment by rpkamp** [comment]
PHP 7.4 was released yesterday. If Travis already supports it than that should be used instead of 7.4snapshot.

File: .travis.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ matrix:
     - php: nightly
     - php: hhvm
   allow_failures:
+    - php: hhvm
```
Comment (stof) [inline .travis.yml]: I would rather remove the job entirely (given that HHVM itself dropped support for PHP interop)

Comment (everzet) [reply .travis.yml]: I was planning to do a major version bump and drop support for all unmaintained PHP versions after this is merged and we do a small minor version bump.

**Comment by Pierstoval** [comment]
What needs to be done to finish this PR? Any help needed?

**Comment by kuraobi** [comment]
Are we waiting for Travis to add PHP7.4 final binary? Is it any different than the latest 7.4snapshot?

**Comment by pfazzi** [comment]
Come on guys, this deprecation warning is still annoying me every time a clear my app cache :smile: 

image

**Comment by kuraobi** [comment]
FYI PHP7.4 is available in Travis CI for Ubuntu 18.04. Is there any reason to stick to 14.04?

File: .travis.yml [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+    - php: 5.3
+      dist: precise
+    - php: 5.4
+    - php: 5.5
     - php: 5.6
+    - php: 7.0
+    - php: 7.1
+    - php: 7.2
+    - php: 7.3
+    - php: 7.4
```
Comment (rpkamp) [inline .travis.yml]:
```suggestion
    - php: 7.4
      dist: bionic
```

That should enable PHP 7.4 as per @kuraobi's comment

Comment (kuraobi) [reply .travis.yml]: Yep, already tested here https://github.com/Behat/Transliterator/pull/30 , everything green

**Comment by kdambekalns** [comment]
Please merge this, PHP 7.4 support would be awesome!

**Comment by snapshotpl** [comment]
@everzet travis now contains stable 7.4 so we are ready to go!

**Comment by Pierstoval** [comment]
I used this in my `composer.json` file:

```diff
    "require": {
...
-        "behat/transliterator": "^1.2",
+        "behat/transliterator": "dev-support-php74 as 1.2.x-dev",
...
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/Behat/Transliterator"
+        }
+    ],
```

My 647-tests suite with good coverage on my biggest project is all green 😉 

All that needs to be fixed is the Travis build, I guess 🤔

**Comment by kdambekalns** [comment]
Small tweak suggestion: Change the PR title to "Support php versions from 5.**6** to 7.4" :)

Also, pleeeeaaaase get this merged and released! ❤️

**Comment by kdambekalns** [comment]
One more comment from me: with a simple `dist:bionic` for the PHP 7.4 build it works, as can be seen on https://travis-ci.org/kdambekalns/Transliterator/builds/621004013

**Comment by alexander-schranz** [comment]
@kdambekalns 7.4 is now also available on trust so just a CI restart should make it green.

**Comment by kuraobi** [comment]
Any news on this? What's preventing the merge of #30 and this one?
For now we use branch aliases but I'm not a fan of having branch aliases in production, and I don't think that's a solution for the 15.8K+ downstream projects to support PHP7.4. So if there is anything else to do, please let us know.

**Comment by kdambekalns** [comment]
🎉

**Comment by kdambekalns** [comment]
Hooray to this being merged, any reason not to release a new version now?

**Comment by TomasVotruba** [comment]
Release would be great, it breaks our build now :/
https://travis-ci.org/KnpLabs/DoctrineBehaviors/jobs/624044738#L371

**Comment by alexander-schranz** [comment]
@stof is there something we can do to help to move the release forward?

**Comment by snapshotpl** [comment]
@everzet @stof missing release of this lib is really annoying for many important projects in php community. Can you move forward this?

**Comment by TomasVotruba** [comment]
Solution: https://github.com/KnpLabs/DoctrineBehaviors/pull/468/files#diff-c5f7d7b9e0f14fa580860143f56f3b7f

Works great :+1:

**Comment by kdambekalns** [comment]
This is getting less funny by the day.

- A lot of libraries (95 says Packagist) out there depend on this package, among them well-known players like (ahem) `behat/behat` and `gedmo/doctrine-extensions`.
- PHP 7.4 has meanwhile seen it's first patch-level release.
- Compatibility is ensured with this PR, which has been merged 10 days ago.
- Multiple people asked for this to get some attention and/or asked how they could help.
- First projects switched to `symfony/strings` instead of this one, even though it still is "experimental"

All that is needed to fix this mess would be **a new release, something that takes three clicks** (counting from https://github.com/Behat/Transliterator) and a tiny bit of typing.

**Please. Help. The. PHP. Community!**

**Comment by nocive** [comment]
@kdambekalns thank you for vocalizing our concerns :bowing_man: :medal_sports:

**Comment by nocive** [comment]
@kdambekalns release has been released! :)

**Comment by kdambekalns** [comment]
> @kdambekalns release has been released! :)

Yes, I know. And we released our depending project as well. So, all set. Just wondering why https://github.com/Behat/Transliterator/issues/25 is still open. But… whatever.

And let's hope the feeling of "staleness" won't come back again…

**Pull request merged**